design/ironic-debuggability-improvement.md

```diff
@@ -0,0 +1,108 @@
+<!--
+ This work is licensed under a Creative Commons Attribution 3.0
+ Unported License.
+
+ http://creativecommons.org/licenses/by/3.0/legalcode
+-->
+
+# ironic-debuggability-improvement
+
+## Status
+
+provisional
+
+## Summary
+
+Ironic and Ironic-inspector logs are currently hidden inside their respective
+containers and never shown to the end user by default.
+This document describes proposed changes in Ironic and Ironic-inspector
+logs handling to make debugging easier for the end user.
+
+## Motivation
+
+We need to make Ironic logs accessible to the end user.
+
+### Goals
+
+1. Make it easier to understand what happened when a deployment fails.
+2. Avoid old logs pile up and take all available space.
+
+### Non-Goals
+
+1. Modify Ironic logging events or log level.
+
+## Proposal
+
+The ironic-inspector-image includes a new script, ironic-inspection-log-watch,
+that can act as an entry point for a container to dump host inspection logs.
+
+The script should watch for Ironic host inspection log files to appear in
+"/shared/logs/ironic-inspector/ramdisk", wait for them to stop changing
```

Comment:
nit, there's a typo in this path, it's `/shared/log`

Reply:
Fixed